client/src/main.rs

```diff
@@ -142,3 +147,41 @@ fn print_help(client_info: &str, commands: &[std::sync::Arc<dyn Command>]) {
     println!("quit | q! \n\tExit this client");
     println!("\n");
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_args_port() {
+        let args = Args::from_iter(&["test", "--host=h", "--validator_set_file=vsf"]);
+        assert_eq!(args.port, 8000);
+        assert_eq!(format!("{}:{}", args.host, args.port), "h:8000");
+        let args = Args::from_iter(&[
+            "test",
+            "--port=65535",
+            "--host=h",
+            "--validator_set_file=vsf",
+        ]);
+        assert_eq!(args.port, 65535);
+    }
+
+    #[test]
+    fn test_args_port_too_large() {
+        let result = Args::from_iter_safe(&[
+            "test",
+            "--port=65536",
+            "--host=h",
+            "--validator_set_file=vsf",
+        ]);
+        assert_eq!(result.is_ok(), false);
+    }
+
+    #[test]
+    fn test_args_port_invalid() {
+        let result =
+            Args::from_iter_safe(&["test", "--port=abc", "--host=h", "--validator_set_file=vsf"]);
+        assert_eq!(result.is_ok(), false);
+    }
+
```

Comment:
And then you could add an explicit test for the Arg parsing like you've done here for the case where a user passes in `0`